### ISSUE TITLE
boot/menu.js Memory leak?

### ISSUE THREAD

**Issue by busbuyukefe** [jeffmer/P8Apps#4]
When I am using menu.js with some app included from [bangleApps](https://github.com/espruino/BangleApps) (alarm and file manager e.g.) the watch stops working and gives error messages about low memory on web IDE.

File Manager app is just modified to include the menu.js but the Alarm app is slightly modified but the part that uses menu.js is untouched, and even if the alarm.js (the part of the app that handles alarm alerts and vibrations) is left empty the problem still continues. I couldn't quite understand what is causing this.
Also strangely enough the alarm app freezes after adding two alarms, after that you can't even click on the entries to delete/edit them or add new alarms. 

![err1](https://user-images.githubusercontent.com/13920840/115108539-b3b86f00-9f79-11eb-9268-9fe27ce2a3a6.png)
![err2](https://user-images.githubusercontent.com/13920840/115108540-b5823280-9f79-11eb-827e-3a3b9c26a184.png)

**Comment by jeffmer** [comment]
Hi, I have not had any problems with menu.js. It is used in Settings and the App manager. Perhaps you can share your file or alarm app code with me and I can have a look.

**Comment by busbuyukefe** [comment]
Yeah sure. Current version of the alarm app just prints something in the console instead  of vibrating.
[alarm.zip](https://github.com/jeffmer/P8Apps/files/6330703/alarm.zip)

**Comment by jeffmer** [comment]
Had a quck look and I can reproduce the low memory error. It seems to be some interaction between widgets and menu. The simple solution for the moment is do not load or draw widgets in apps that use menu. When I removed the lines in your code which load and draw widgets, I created 5 alarms with no problems. Not sure that it is much of a problem not to see widgets at the sane time as menu as normally the app is only active for setting alarms.
